jupyterkernelgen/jupyterkernelgen.py

```diff
@@ -0,0 +1,278 @@
+#!/usr/bin/env python3
+import os
+import glob
+import shutil
+import sys
+import readline
+import subprocess
+
+# allow tab completion in input()
+readline.set_completer_delims(' \t\n=')
+readline.parse_and_bind("tab: complete")
+
+class JupyterKernelGenException(BaseException):
+    pass
+
+# define ANSI escape codes for coloured text
+class colors:
+    HEADER = '\033[95m'
+    OK = '\033[92m'
+    WARNING = '\033[93m'
+    FAIL = '\033[91m'
+    ENDC = '\033[0m'
+
+def clean_exit(exit_code: int, path: str | None = None) -> None:
+    """
+    Exit the program and remove the created directory if applicable
+
+    :param exit_code:   the exit code to pass to the operating system
+    :param path:        the path to remove if applicable
+    """
+    try:
+        if path != None:
+            shutil.rmtree(path)
+    except shutil.Error as e:
+        print(f"{colors.FAIL}failed to remove directory: {e}{colors.ENDC}")
+
+    sys.exit(exit_code)
+
+def check_for_conda() -> str:
+    """
+    Looks for a mamba or conda executable on the system with a preference
+    for mamba
+
+    :returns: the path to the mamba or conda executable
+    """
+    print(f"{colors.HEADER}LOOKING FOR CONDA...{colors.ENDC}")
+
+    conda_exe = None
+    try:
+        conda_exe = shutil.which("mamba") # first try getting mamba because its faster
+    except shutil.Error as e:
+        print(f"{colors.FAIL}error occurred getting mamba from path: {e}{colors.ENDC}", file=sys.stderr)
+
+    if conda_exe == None:
+        try:
+            conda_exe = shutil.which("conda") # try finding conda if mamba isn't available
+        except shutil.Error as e:
+            print(f"{colors.FAIL}error occurred getting conda from path: {e}{colors.ENDC}", file=sys.stderr)
+            raise JupyterKernelGenException
+
+        if conda_exe == None:
+            print(f"{colors.FAIL}no conda executable on the path. Exiting...{colors.ENDC}", file=sys.stderr)
+            raise JupyterKernelGenException
+
+    print(f"- found conda executable: {colors.OK}{conda_exe}{colors.ENDC}\n")
+
+    return conda_exe
+
+def get_conda_env():
+    """
+    Gets a path to a conda environment from the user.
+
+    :returns: the path to the conda environment
+    """
+    try:
+        # replace ~ with user's home directory and make it an absolute path
+        # if ~ is not used to refer to the user's home directory, it will
+        # just create an absolute path to the specified location
+        conda_env = os.path.abspath(os.path.expanduser(input(f"{colors.HEADER}ENTER PATH TO CONDA ENVIRONMENT{colors.ENDC}: ")))
```

Comment:
Perhaps not all caps for output to user since some may think you are yelling.
Add wording that you need FULL PATH and give an example `/home/user/.conda/env/fastqc`